### PR TITLE
Fixing ValidateCredentialsAsync when running under netcore or running…

### PR DESCRIPTION
**Pull request by NatMarchand**
… behind a proxy

See issue #1499 for details

### CODE REVIEW

**Comment by NatMarchand** [comment]
Is there anything I can do to make the build pass ?
I don't think it's my code which broke the build.

File: src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs [hunk 1]
```diff
@@ -71,21 +69,30 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
...
-                throw;
+                // Inner exception when using WindowsAzure.Storage for net45
+                case WebException webException:
+                    return webException.Status == WebExceptionStatus.NameResolutionFailure;
+                // Inner exception when using WindowsAzure.Storage for netstandard1.3 as WebException does not exist
+                case System.Net.Http.HttpRequestException httpRequestException:
+                    return (httpRequestException.HResult & 0xFFFF) == errorWinhttpNameNotResolved;
+                // When running behind a proxy
+                case null:
+                    return storageException.RequestInformation?.HttpStatusCode == serviceUnavailable;
```
Comment (paulbatum) [inline src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs]: Can you explain this case? Inferring a blob-only storage account from something as generic as a 503 doesn't seem right but perhaps I'm missing something.

**Comment by paulbatum** [comment]
@NatMarchand there's an issue with our build of public PRs (and therefore with no secrets) for v2. We'll be able to confirm these changes are good by doing a full test run (with secrets).

@brettsam can you review this PR?

File: src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs (hunk 1 above)
Comment (NatMarchand) [reply src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs]: In fact, I've tried this code at home and at work. At work, we're behind a corporate http proxy (squid) which "catch" the resolution error and gives a 503. Not sure if it's specific to squid or all the proxy. Maybe we can go without ?

**Comment by NatMarchand** [comment]
Hi!
Were you able to run the tests for this PR ?
Regards,

File: src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs (hunk 1 above)
Comment (paulbatum) [reply src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs]: @NatMarchand This doesn't seem resolved. I still have issues with inferring a blob-only account from a generic 503...

Comment (NatMarchand) [reply src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs]: Ok, will remove the case when rebasing then :)

**Comment by NatMarchand** [comment]
Rebased and rewritten to remove the 503 case 😀

**Pull request merged**